This patch is based on the upstream commit described below, adapted
for use in the Debian package by Peter Michael Green

commit b4a73ef44d5b7e84584c784d8865553774dfd71d
Author: Andrew Walbran <qwandor@google.com>
Date:   Tue Jun 7 14:13:02 2022 +0100

    Update to nix 0.24.1.
    
    This makes the contents of VsockAddr private, so we have to use the
    provided methods instead.
    
    Signed-off-by: Andrew Walbran <qwandor@google.com>

Index: vsock/src/lib.rs
===================================================================
--- vsock.orig/src/lib.rs
+++ vsock/src/lib.rs
@@ -56,7 +56,7 @@ impl VsockListener {
     /// Create a new VsockListener which is bound and listening on the socket address.
     pub fn bind(addr: &SockAddr) -> Result<VsockListener> {
         let mut vsock_addr = if let SockAddr::Vsock(addr) = addr {
-            addr.0
+            addr.as_ref().to_owned()
         } else {
             return Err(Error::new(
                 ErrorKind::Other,
@@ -114,7 +114,10 @@ impl VsockListener {
         {
             Err(Error::last_os_error())
         } else {
-            Ok(SockAddr::Vsock(VsockAddr(vsock_addr)))
+            Ok(SockAddr::Vsock(VsockAddr::new(
+                vsock_addr.svm_cid,
+                vsock_addr.svm_port,
+            )))
         }
     }
 
@@ -227,7 +230,7 @@ impl VsockStream {
     /// Open a connection to a remote host.
     pub fn connect(addr: &SockAddr) -> Result<Self> {
         let vsock_addr = if let SockAddr::Vsock(addr) = addr {
-            addr.0
+            addr.as_ref()
         } else {
             return Err(Error::new(
                 ErrorKind::Other,
@@ -242,7 +245,7 @@ impl VsockStream {
         if unsafe {
             connect(
                 sock,
-                &vsock_addr as *const _ as *const sockaddr,
+                vsock_addr as *const _ as *const sockaddr,
                 size_of::<sockaddr_vm>() as socklen_t,
             )
         } < 0
@@ -278,7 +281,10 @@ impl VsockStream {
         {
             Err(Error::last_os_error())
         } else {
-            Ok(SockAddr::Vsock(VsockAddr(vsock_addr)))
+            Ok(SockAddr::Vsock(VsockAddr::new(
+                vsock_addr.svm_cid,
+                vsock_addr.svm_port,
+            )))
         }
     }
 
@@ -302,7 +308,10 @@ impl VsockStream {
         {
             Err(Error::last_os_error())
         } else {
-            Ok(SockAddr::Vsock(VsockAddr(vsock_addr)))
+            Ok(SockAddr::Vsock(VsockAddr::new(
+                vsock_addr.svm_cid,
+                vsock_addr.svm_port,
+            )))
         }
     }
 
Index: vsock/Cargo.toml
===================================================================
--- vsock.orig/Cargo.toml
+++ vsock/Cargo.toml
@@ -27,3 +27,3 @@ repository = "https://github.com/rust-vs
 [dependencies.nix]
-version = "0.19.1"
+version = "0.26"
 [dev-dependencies.rand]
